reconciler/events_test.go

```diff
@@ -0,0 +1,107 @@
+/*
+Copyright 2019 The Knative Authors
+
+Licensed under the Apache License, Veroute.on 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package reconciler
+
+import (
+	"errors"
+	"io"
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+	corev1 "k8s.io/api/core/v1"
+)
+
+const (
+	exampleStatusFailed = "ExampleStatusFailed"
+)
+
+func TestNil_Is(t *testing.T) {
+	var err error
+	if EventIs(err, NewEvent(corev1.EventTypeWarning, exampleStatusFailed, "")) {
+		t.Error("did not expect error to be a ReconcilerEvent")
+	}
+}
+
+func TestError_Is(t *testing.T) {
+	err := errors.New("some other error")
+	if EventIs(err, NewEvent(corev1.EventTypeWarning, exampleStatusFailed, "")) {
+		t.Error("did not expect error to be a ReconcilerEvent")
+	}
+}
+
+func TestNew_Is(t *testing.T) {
+	err := NewEvent(corev1.EventTypeWarning, exampleStatusFailed, "this is an example error, %s", "yep")
+	if !EventIs(err, NewEvent(corev1.EventTypeWarning, exampleStatusFailed, "")) {
+		t.Error("expected error to be a ReconcilerEvent, is not")
+	}
+}
+
+func TestNewOtherType_Is(t *testing.T) {
+	err := NewEvent(corev1.EventTypeNormal, exampleStatusFailed, "this is an example error, %s", "yep")
+	if EventIs(err, NewEvent(corev1.EventTypeWarning, exampleStatusFailed, "")) {
+		t.Error("not a Warn, ExampleStatusFailed")
```

Comment:
```suggestion
		t.Error("Not a Warn, ExampleStatusFailed")
```
etc, won't comment on the rest :)